### PR TITLE
Add longer descriptions to tests to verify multi-line behavior

### PR DESCRIPTION
**Pull request by Leonidas-from-XIV**
Adds test cases that test the behavior of `\` in descriptions.

Requires #7401.

### CODE REVIEW

File: test/blackbox-tests/test-cases/dune-project-meta/basic-generate.t [hunk 1]
```diff
@@ -29,12 +29,28 @@ The `dune build` should generate the opam file
...
   >   (depends
   >     (cohttp (>= 1.0.2))
   >     (conduit-async (>= 1.0.3))
   >     (async (>= v0.10.0))
   >     (async (< v0.12))))
+  > ;
+  > (package
+  >   (name cohttp-lwt)
+  >   (synopsis "HTTP client and server for the Lwt library")
+  >   (description "
```
Comment (emillon) [inline test/blackbox-tests/test-cases/dune-project-meta/basic-generate.t:45]: is it intended that the first character of the description is a newline?

Comment (Leonidas-from-XIV) [reply test/blackbox-tests/test-cases/dune-project-meta/basic-generate.t:45]: Yes. This is taken from from the regression in dune generating it's own opam files that Rudi reported, where the description in `dune.opam` used to start with an empty line and doesn't anymore.

I think the new way is sort of nicer but we probably don't want all OPAM files to be reformatted as a consequence of us bumping the version of our vendored o-f-f.

Comment (emillon) [reply test/blackbox-tests/test-cases/dune-project-meta/basic-generate.t:45]: ok. can you update the string explaining that? (or at least change the "multi line" part to "a string that starts with a newline")

File: test/blackbox-tests/test-cases/dune-project-meta/basic-generate.t [hunk 2]
```diff
@@ -29,12 +29,28 @@ The `dune build` should generate the opam file
...
+  >   (description "\
+  > A _really_long multi line description that spans across lines to
+  > make sure that the rendering of long strings stays compatible.
+  > ")
   >   (depends
   >     (cohttp (>= 1.0.2))
   >     (conduit-async (>= 1.0.3))
   >     (async (>= v0.10.0))
   >     (async (< v0.12))))
+  > ;
```
Comment (emillon) [inline test/blackbox-tests/test-cases/dune-project-meta/basic-generate.t:41]: nitpick but is this comment necessary?

Comment (Leonidas-from-XIV) [reply test/blackbox-tests/test-cases/dune-project-meta/basic-generate.t]: No. The other lines had these comments so for consistency I added it. But it is not required so to be consistent I removed it from all lines.

File: test/blackbox-tests/test-cases/dune-project-meta/basic-generate.t (hunk 1 above)
Comment (Leonidas-from-XIV) [reply test/blackbox-tests/test-cases/dune-project-meta/basic-generate.t:45]: Done; I hope it is clearer now.

**Review by emillon — APPROVED** [review]
ah ok I didn't see the `;` above. You can also leave it I suppose. I'll leave the decision to you, either's fine

**Comment by Leonidas-from-XIV** [comment]
I think it is better without these comments, since it is less confusing. For a moment I thought they are separators of some sort since there was no text to be commented.

**Comment by emillon** [comment]
ok let's go!

**Pull request merged**